lib/matplotlib/widgets.py

```diff
@@ -2966,100 +3057,127 @@ def _release(self, event):
     def _onmove(self, event):
         """Motion notify event handler."""
 
-        state = self._state | self._default_state
+        state = self._state
+        rotate = ('rotate' in state and
+                  self._active_handle in self._corner_order)
+        eventpress = self._eventpress
+        # The calculations are done for rotation at zero: we apply inverse
+        # transformation to events except when we rotate and move
+        if not (self._active_handle == 'C' or rotate):
+            inv_tr = self._get_rotation_transform().inverted()
+            event.xdata, event.ydata = inv_tr.transform(
+                [event.xdata, event.ydata])
+            eventpress.xdata, eventpress.ydata = inv_tr.transform(
+                [eventpress.xdata, eventpress.ydata]
+                )
+
+        dx = event.xdata - eventpress.xdata
+        dy = event.ydata - eventpress.ydata
+        # refmax is used when moving the corner handle with the square state
+        # and is the maximum between refx and refy
+        refmax = None
+        if self._use_data_coordinates:
+            refx, refy = dx, dy
+        else:
+            # Add 1e-6 to avoid divided by zero error
+            refx = event.xdata / (eventpress.xdata + 1e-6)
+            refy = event.ydata / (eventpress.ydata + 1e-6)
```

Comment:
```suggestion
        # refmax is used when moving the corner handle with the square state
        # and is the maximum between refx and refy
        refmax = None
        if self._use_data_coordinates:
            refx = event.xdata - eventpress.xdata
            refy = event.ydata - eventpress.ydata
        else:
            refx = event.xdata * eventpress.ydata
            refy = event.ydata * eventpress.xdata
```
Reopening this discussion, as
1. `dx`, `dy` are never used again in this block; writing the definitions in a more symmetric fashion should clarify a bit better what is happening in the two cases.
2. The padding will not completely protect from division by zero, as in principle `eventpress.xdata` may also be -1e-6, however unlikely that would be. As the absolute values of `refx`, `refy` are not used anywhere below, only their relative amounts, multiplying both by both denominators would avoid this.
3. Looking a bit beyond that, the values themselves are not really used at all, except to determine which of `abs(refx)`, `abs(refy)` is larger (via the float comparison `refmax == refx`, which in itself feels a bit problematic).
So it seems all that's really needed here is a single bool or other indicator to tell which axis displacement to use for the `'square'` case.

I think all the logic below could be combined in
```python
        # `use_x` indicates whether to use displacement in `x` or `y` for setting
        # the size of a 'square' state (larger of both when moving a corner handle).
        use_x = False
        if self._active_handle not in ['S', 'N']:
            if self._active_handle in ['E', 'W']:
                use_x = True
            elif self._use_data_coordinates:
                if (abs(event.xdata - eventpress.xdata) >
                        abs(event.ydata - eventpress.ydata)):
                    use_x = True
            elif (abs(event.xdata * eventpress.ydata) >
                     abs(event.ydata * eventpress.xdata)):
                use_x = True
```
I guess it's still a matter of personal preference if you find it clearer to keep the `if self._active_handle in ...` in the individual blocks below.
It is also not fully clear to me what precisely the `event.xdata` / `eventpress.xdata` return, respectively, so I have not fully understood the rationale for using the difference for the `_use_data_coordinates` case, and a scaled absolute location otherwise. But it looks like it produces sensible behaviour for the square options either way.

Reply:
The first part would also address @timhoffm's note in https://github.com/matplotlib/matplotlib/pull/20839#discussion_r762612282; the rest may indeed delay this one too much by requiring another round of reviews.